src/azure-cli-core/azure/cli/core/error_assistance.py

```diff
@@ -0,0 +1,171 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import json
+import shutil
+import requests
+
+from typing import Union
+from azure.cli.core import AzCli
+from azure.cli.core.style import Style, print_styled_text
+from knack.log import get_logger
+
+logger = get_logger(__name__)
+
+# TODO: These values are placeholders now.
+# Will need to replace them with the actual values when it's ready in production.
+_DEEPPROMPT_ENDPOINT = "<url>"
+_DEEPPROMPT_APP = "<appid>"
+_AAD_TENANT = "<tenant>"
+_SCOPES = [f"{_DEEPPROMPT_APP}/.default"]
+_TIMEOUT = 180
+
+_cached_token_session: tuple = ()
+
+
+def request_error_assistance(command: Union[str, None] = None,
+                             error: Union[str, None] = None,
+                             cli_ctx: Union[AzCli, None] = None) -> dict:
+    if not _error_enabled(cli_ctx):
+        return {}
+
+    if not command:
+        logger.error("command cannot be empty.")
+        return {}
```

Comment:
It seems that the `command` could be `None` for `KeyboardException` in `src/azure-cli-core/azure/cli/core/util.py` line 155 to line 157.